app/Console/Commands/CheckSchemaCommand.php

```diff
@@ -0,0 +1,462 @@
+        'name',
+        'total_budget',
+        'recipient_organization_budget',
+        'recipient_region_budget',
+        'recipient_country_budget',
+        'total_expenditure',
+        'document_link',
+    ];
+
+    /**
+     * @var array|string[]
+     */
+    protected array $settingDataSchema = [
+        'registry_info',
+        'default_field_values',
+        'default_field_groups',
+        'simplify_fields',
+    ];
+
+    /**
+     * @var array|string[]
+     */
+    protected array $activityDataSchema = [
+        'identifier',
+        'other_identifier',
+        'title',
+        'description',
+        'activity_date',
+        'contact_info',
+        'participating_organization',
+        'recipient_country',
+        'recipient_region',
+        'location',
+        'sector',
+        'country_budget_items',
+        'humanitarian_scope',
+        'policy_marker',
+        'default_aid_type',
+        'budget',
+        'planned_disbursement',
+        'document_link',
+        'related_activity',
+        'conditions',
+        'default_field_values',
+        'guidelines',
+    ];
+
+    /**
+     * @var array|string[]
+     */
+    protected array $activityTransactionDataSchema = [
+      'transaction',
+    ];
+
+    /**
+     * @var array|string[]
+     */
+    protected array $baseResultDataSchema = [
+        'title',
+        'description',
+    ];
+
+    /**
+     * @var array|string[]
+     */
+    protected array $resultDocumentLinkDataSchema = [
+        'title',
+        'description',
+        'category',
+        'language',
+    ];
+
+    /**
+     * @var
+     */
+    protected $spreadSheet;
+
+    /**
+     * @var array
+     */
+    protected array $logStack = [];
+
+    /**
+     * @return void
+     * @throws \JsonException
+     * @throws Exception
+     */
+    public function checkSchema(): void
+    {
+        $this->spreadSheet = new Spreadsheet();
+        $this->checkOrganizationDataSchema();
+        $this->checkSettingDataSchema();
+        $this->checkActivityDataSchema();
+        $this->checkBaseResultDataSchema();
+        $this->checkActivityTransactionDataSchema();
+        $this->generateExcelFile();
+        $xlsFile = new Xlsx($this->spreadSheet);
+        $fileName = app_path('DataMigration/remarks.xlsx');
+        $xlsFile->save($fileName);
+    }
+
+    /**
+     * @return void
+     * @throws \JsonException
+     */
+    public function checkOrganizationDataSchema(): void
+    {
+//        $this->spreadSheet->createSheet();
+        $this->spreadSheet->setActiveSheetIndex(0);
+        $this->spreadSheet->getActiveSheet()->setTitle('organization_data');
+
+        $aidStreamOrganizationDataTemplate = readJsonFile('DataMigration/Templates/AidStreamOrganizationDataSchema.json');
+        $aidStreamOrganizationData = $this->db::connection('aidstream')
+            ->table('organization_data')
+            ->whereIn('organization_id', $this->organizationIds)
```

Comment:
if this is for the data that will be migrated, the migration command uses an extra check is_reporting_org => true